include/inexor/vulkan-renderer/wrapper/cpu_texture.hpp

```diff
@@ -0,0 +1,75 @@
+﻿#pragma once
+
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+class CpuTexture {
+private:
+    const std::string m_file_name;
+    const std::string m_name;
+
+    int m_texture_width{0};
+    int m_texture_height{0};
+    int m_texture_channels{0};
+    int m_mip_levels{0};
+
+    std::vector<unsigned char> m_texture_data;
+
+    ///
+    void read_texture();
+
+    ///
+    void generate_error_texture_data();
+
+public:
+    /// @brief Creates a CpuTexture instance with a default texture.
+    CpuTexture();
+
+    /// @brief Reads a texture from a file.
+    /// @param file_name [in] The file name of the texture.
+    /// @param name [in] The internal memory allocation name of the texture.
+    CpuTexture(const std::string &file_name, const std::string &name);
+
+    CpuTexture(const CpuTexture &) = delete;
+    CpuTexture(CpuTexture &&) noexcept;
+    ~CpuTexture();
+
+    CpuTexture &operator=(const CpuTexture &) = delete;
+    CpuTexture &operator=(CpuTexture &&) = default;
+
+    [[nodiscard]] const std::string &file_name() const {
+        return m_file_name;
+    }
+
+    [[nodiscard]] const std::string &name() const {
+        return m_name;
+    }
+
+    [[nodiscard]] const int width() const {
+        return m_texture_width;
+    }
+
+    [[nodiscard]] const int height() const {
+        return m_texture_height;
+    }
+
+    [[nodiscard]] const int channels() const {
+        return m_texture_channels;
+    }
+
+    [[nodiscard]] const int mip_levels() const {
+        return m_mip_levels;
+    }
+
+    [[nodiscard]] void *data() const {
+        return const_cast<void *>(static_cast<const void *>(m_texture_data.data()));
+    }
+
+    [[nodiscard]] const std::size_t data_size() const {
+        return m_texture_data.size();
+    }
```

Comment:
Do not return as `const`

Reply:
Fixed.  I also updated the equivalents in GpuTexture.